### DISCUSSION TITLE
How can I set allow-tray in globals.local instead of firejail.config?

### DISCUSSION THREAD

**Discussion by amano-kenji** [netblue30/firejail#5173] [Q&A]
I don't like modifying /etc/firejail/firejail.config which a linux distribution tries overwrite during each upgrade.
I don't feel comfortable with setting anything in firejail.config.

I want to set allow-tray in globals.local. How can I do that?

**Comment by rusty-snake** [comment]
> I want to set allow-tray in globals.local. How can I do that?

You can't.

firejail.config is the firejail configuration file.
globals.local is a profile file.

Nevertheless it would be really useful if you could set conditions like ALLOW_TRAY or BROWSER_DISABLE_U2F also in profiles (locals).

> I don't like modifying /etc/firejail/firejail.config which a linux distribution tries overwrite during each upgrade.

Use a better distro :speak_no_evil: :slightly_smiling_face:.

You can try `chattr +i /etc/firejail/firejail.config`.

---

OT: We should add `allow-tray` to firejail-welcome.sh.

**Reply by amano-kenji** [reply]
Every linux distribution tries to rewrite configuration files in /etc during each upgrade. Programs learned to work around this by providing a configuration directory where any config file is parsed by the programs.

Arch linux deals with it through pacdiff. Gentoo linux has dispatch-conf. I have become tired of merging old config files with new ones in /etc.

A config directory works around that.

If firejail read config files in /etc/firejail/config/, I wouldn't need to worry about merging new versions of firejail.config with old ones.

**Reply by amano-kenji** [reply]
The ability to set conditions like ALLOW_TRAY in globals.local would be really appreciated. It is better than /etc/firejail/config/.

Something like

```
set-condition ALLOW-TRAY yes
```

**Comment by ghost** [comment]
To summarize the above: Firejail could be more user-friendly by offering a way to persistently override conditionals, per profile and/or globally. I haven't looked at the code in detail yet from this perspective, but a `firejail.local` (override functionality for /etc/firejail/firejail.config) might accomodate both?

**Reply by amano-kenji** [reply]
Something like that.

**Reply by rusty-snake** [reply]
Moving the parsing from https://github.com/netblue30/firejail/blob/880f2c98a1dee26228530875fc45d54db68ed1c8/src/firejail/checkcfg.c#L41 to an own function and do an

```C
read_cfg(SYSCONFDIR "/firejail.config");
read_cfg(SYSCONFDIR "/firejail.local");
```

so if anyone wants to learn a bit about firejails codebase ...

**Reply by amano-kenji** [reply]
Please add something like firejail.local.
